raiden-ts/src/raiden.ts

```diff
@@ -684,13 +706,19 @@ export class Raiden {
    *                - secret  Secret to register, a random one will be generated if missing
    *                - secrethash  Must match secret, if both provided, or else, secret must be
    *                              informed to target by other means, and reveal can't be performed
+   *                - metadata  Used to specify possible routes instead of querying PFS.
    * @returns A promise to transfer's secrethash (unique id) when it's accepted
    */
   public async transfer(
     token: string,
     target: string,
     amount: BigNumberish,
-    options: { paymentId?: BigNumberish; secret?: string; secrethash?: string } = {},
+    options: {
+      paymentId?: BigNumberish;
+      secret?: string;
+      secrethash?: string;
+      metadata?: { readonly routes: readonly { route: readonly string[] }[] };
```

Comment:
```suggestion
      metadata?: Metadata
```

Reply:
We can't do that: `Metadata` uses the internal, strong/branded types (e.g. `Address`), which either the user need to know about and do the proper validation process, or we expose these "simplified" versions of the types/schemas (in this case, it's like Metadata but using the raw strings instead of Address), and then validate inside, as I did right below, and from there on, it's recognized as actually being `Metadata`. The same happens for all Addresses in the public methods.

Reply:
We use `RaidenChannel` in the dapp, which also uses `Address`. User can validate with `ethers.utils.getAddress()` can't they?

Reply:
`RaidenChannel` is an output. There, it's safe to use specific types, as any of them can be assigned to just the primitives as usual. Inputs can't be validated only in the usual way (like `getAddress`) to narrow the type, so if the dApp needs to know about and pass an actual `Address`, they need to import it and use things like `Address.is` or `Address.decode` and then pass it.
If we accept the primitives, we just validate it internally and throw if not valid, and users don't need to know about the specific/branded/internal types, as outputs are safely upcasted.

Reply:
Ok cool make sense, thanks for clarifying